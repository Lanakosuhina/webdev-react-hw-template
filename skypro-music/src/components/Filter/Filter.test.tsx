import renderer from "react-test-renderer";
import { Filter } from ".";

describe('Компонент фильтра', () => {
  const mockToggleSelected = jest.fn();
  const mockOnClick = jest.fn();

  it('корректно отрисовывает разметку с начальными значениями isOpen = true', () => {
    const component = renderer.create(<Filter
      list={['item 1', 'item 2']}
      title={"жанру"}
      isOpen={true}
      toggleSelected={mockToggleSelected}
      onClick={mockOnClick}
      counter={0}
    />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('корректно срабатывает обработчик клика', () => {
    const component = renderer.create(<Filter
      list={[]}
      title={"жанру"}
      isOpen={true}
      toggleSelected={mockToggleSelected}
      onClick={mockOnClick}
      counter={0}
    />);

    let button = component.root.findByType("button");
    button.props.onClick();
    expect(mockOnClick).toHaveBeenCalled();
  });
});
