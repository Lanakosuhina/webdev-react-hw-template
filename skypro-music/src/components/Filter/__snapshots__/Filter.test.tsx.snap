// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент фильтра корректно отрисовывает разметку с начальными значениями 1`] = `
<div
  className="wrapper"
>
  <button
    className="filterButton btnText"
    onClick={[MockFunction]}
  >
    Жанры
  </button>
</div>
`;

exports[`Компонент фильтра корректно отрисовывает разметку с начальными значениями isOpen = true 1`] = `
<div
  className="wrapper"
>
  <button
    className="filterButton btnText"
    onClick={[MockFunction]}
  >
    жанру
  </button>
  <ul
    className="dropdown"
  >
    <div
      className="wrap"
    >
      <li
        className="list"
        onClick={[Function]}
      >
        item 1
      </li>
      <li
        className="list"
        onClick={[Function]}
      >
        item 2
      </li>
    </div>
  </ul>
</div>
`;
